tests/ztest/mock_fff/testcase.yaml

```diff
@@ -1,11 +1,9 @@
 tests:
   testing.ztest.mock_fff:
-    filter: CONFIG_ZEPHYR_FFF_MODULE
```

Comment:
It's causing https://github.com/zephyrproject-rtos/zephyr/pull/36433 to break. Since the test is always being skipped. I don't think that the group `ci` is working.

Reply:
It's causing https://github.com/zephyrproject-rtos/zephyr/pull/36433 to break, because, in the current form it is expected that these tests will be executed by CI on-commit on native_posix (because native_posix is part of integration_platforms in this testcase.yaml).
So the main question is: Are those tests expected to be executed by on-commit CI. If not, integration_platforms will have to be removed. However, If this is the case, keep in mind there won't be any warranty that they will be verified anywhere else.

Reply:
@nashif the issue is that as is, the tests were not being run in the CI.

Reply:
Quick summary to make sure we're all on the same page:

The current setup uses a group filter `ci` to make sure that we only run these tests on the CI. But, the tests are not actually being run. That's what's causing #36433 to fail. This PR makes it such that the tests _should_ run, but I'm still running into issues where twister is not seeing the FFF module. Even if I run `west update --gf +ci`, twister isn't seeing the module. But, if I remove the group filter from `west.yml`, then these tests pass.

What I've found is that when the group filter is present in `west.yml`, then the generated `Kconfig.modules` does not include FFF. This is the source of the issue. I haven't been able to find a way to tell twister to run with the `ci` environment.